File: Week_1_C/agpollackberti-code-cs50-problems-2020-x-cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //get and shape user input from float to int
    float changeOwed;
    do
    {
        changeOwed = get_float("Change owed: ");
    }
    while (changeOwed <= 0 || changeOwed > 99);    
    
    int intOwed = round(changeOwed * 100);

    //initialize counters
    int remainingAmount = intOwed; 
    int coinCounter = 0;

    //now start getting greedy!
    //can I use quarters?
    if (remainingAmount >= 25)
    {
        do
        {
            remainingAmount = (remainingAmount - 25);
            coinCounter++;
        }   
        while (remainingAmount >= 25);
    }
    //can I use dimes?
    if (remainingAmount >= 10)
    {
        do
        {
            remainingAmount = (remainingAmount - 10);
            coinCounter++;
        }
        while (remainingAmount >= 10);
    }
    //can I use nickels?
    if (remainingAmount >= 5)
    {
        do
        {
            remainingAmount = (remainingAmount - 5);
            coinCounter++;
        }
        while (remainingAmount >= 5);
    }
    //can I use pennies?
    if (remainingAmount >= 1)
    {
        do
        {
            remainingAmount = (remainingAmount - 1);
            coinCounter++;
        }
        while (remainingAmount >= 1);
    }

    //once we've exhausted remainingAmount, then return the count
    printf("%i \n", coinCounter);

}

File: Week_6_Python/agpollackberti-code-cs50-problems-2020-x-sentimental-cash/cash.py
# import files
from cs50 import get_float

# INPUT
# get user input
while True:
    amountfloat = get_float("Change owed: ")
    if amountfloat >= 0:
        break
# turn float to int to get nearest cent
amount = round(amountfloat*100)

# CALCULATION
coinCount = 0

# quarters
while (amount >= 25):
    coinCount = coinCount + 1
    amount = amount - 25

# dimes
while (amount >= 10):
    coinCount = coinCount + 1
    amount = amount - 10

# nickles
while (amount >= 5):
    coinCount = coinCount + 1
    amount = amount - 5

# pennies
while (amount >= 1):
    coinCount = coinCount + 1
    amount = amount - 1

# OUTPUT
# return min # of coins required to give user change
print(coinCount)

